### PR TITLE
Chore: fix app inconsistencies

### PR DESCRIPTION
**Pull request by fredshema**
### What does this PR do?

This PR includes fixes to designs, implementations and code style that have accumulated in the repo.

### Description of Task to be completed?

- [x] Redirect trainee to dashboard if already authenticated
- [x] Redirect trainee to login if no token is found when in dashboard 
- [x] Refactor login flow and redesign its inputs for better accessibility
- [x] Remove duplicate GET_PROFILE queries
- [x] Display correct user avatar in navbar
- [x] Refactor logout logic to prevent back taking you to dashboard 

### How should this be manually tested?

Check the list of task completed and test all features associated with them,

### CODE REVIEW

**Review by Mbabazi12 — APPROVED** [review]

**Review by MugemaneBertin2001 — APPROVED** [review]

**Comment by ncutixavier** [comment]
@fredshema  & @MugemaneBertin2001 , This PR is missing screenshots of changed pages

**Comment by fredshema** [comment]
> @fredshema & @MugemaneBertin2001 , This PR is missing screenshots of changed pages

@ncutixavier screenshots are added

File: app/dashboard/trainee/profile.tsx [hunk 1]
```diff
@@ -94,30 +81,34 @@ export default function Profile() {
                 profile.avatar
                   ? { uri: profile.avatar }
                   : {
-                      uri: `https://ui-avatars.com/api/?name=${encodeURIComponent(profile.name)}&background=random&color=fff`,
+                      uri: `https://ui-avatars.com/api/?name=${encodeURIComponent(profile?.name || 'Unknown')}&background=cbecd7&size=256`,
```
Comment (ncutixavier) [inline app/dashboard/trainee/profile.tsx]: Let's remove all unnecessary third parties links

File: app/dashboard/trainee/profile.tsx [hunk 2]
```diff
@@ -94,30 +81,34 @@ export default function Profile() {
...
                     }
               }
-              className="w-[100] h-[100] rounded-full"
+              className="w-28 h-28 rounded-full"
             />
-            <TouchableOpacity className="absolute right-[-65px] bottom-8 p-1">
-              <SvgXml xml={editPic} />
+            <TouchableOpacity
```
Comment (ncutixavier) [inline app/dashboard/trainee/profile.tsx]: Refactor this route

Suggestions: `/dashboard/edit-profile` or `/dashboard/profile/edit`

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**